packages/core/src/app.ts

```diff
@@ -119,11 +119,44 @@ export class Index extends Serializable {
         return this.items.filter((item) => item.hosts.some((h) => h === hashedHost)).length;
     }
 
+    splitHostBySubdomains(host: string) {
+        const parts = host.split(".");
+
+        // Ignore single domains
+        if (parts.length <= 2) {
+            return [];
+        }
+
+        // Remove the tld and domain from the parts, build it separately
+        const domain: string[] = [];
+
+        domain.unshift(parts.pop()!);
+        domain.unshift(parts.pop()!);
+
+        // Build list of subdomains to match, so given 'login.accounts.google.com', we'd see ['login', 'accounts'] as parts and ['google', 'com'] as domain, which should return ['google.com', 'accounts.google.com', and 'login.accounts.google.com']
```

Comment:
This comment is mainly here to help quickly test the lines of code below, but I also think it makes it clearer as to what's happening. I've always disliked how `reduce` makes code less understandable, at least for me, but its' the best option, here, really.